profiler/profiler.go

```diff
@@ -154,6 +158,15 @@ func (p *profiler) run() {
 	if _, ok := p.cfg.types[BlockProfile]; ok {
 		runtime.SetBlockProfileRate(p.cfg.blockRate)
 	}
+	if s, ok := p.cfg.statsd.(statsdGauge); ok {
+		// If the statsd client supports gauges (e.g.
+		// github.com/datadog/datadog-go/v5/statsd) then we can send
+		// gauges reporting the current profiler configuration
+		s.Gauge("datadog.profiler.go.profile_period", p.cfg.period.Seconds(), nil, 1)
+		s.Gauge("datadog.profiler.go.cpu_duration", p.cfg.cpuDuration.Seconds(), nil, 1)
+		s.Gauge("datadog.profiler.go.block_profile_rate", float64(p.cfg.blockRate), nil, 1)
+		s.Gauge("datadog.profiler.go.mutex_profile_fraction", float64(p.cfg.mutexFraction), nil, 1)
+	}
```

Comment:
NIT: Maybe add `runtime.MemProfileRate` as well and the effective value of `WithDeltaProfiles()` (0 or 1) as well?